### ISSUE TITLE
mingw-x64-x86_64-curl

### ISSUE THREAD

**Issue by podaen** [kyaulabs/win11tweak#18]
Hi,

Everything is working fine. Had to do a an uninstall and rebegins on an install from 6 months ago, but I am as far this

![ssl error](https://github.com/kyaulabs/win11tweak/assets/63886347/253125a1-4d77-4be9-9486-1b8d8c6b3698)

Could you insert a pause on the removal to press [Y] please. I would appreciate it!

**Comment by podaen** [comment]
I looked up to where the "mingw-w64-x86_64-curl-winssl" was located.

For me it was...

C:\msys64\var\lib\pacman\local\mingw-w64-x86_64-curl-winssl-8.2.1-1

I deleted it and did the CMD command over and this time I finished successfully. 

Had some other issue with ssl to day! Maybe this will be solved too. Thanks for that.

-------------------------------------------------------------------


On the installation process I have a few remarks.

I would install HWinfo64, HW monitor, Heavenbenchmark, OCCT, prime94 and furmark for stability testing. 

Agave have a lot of errors during the installations. I would remove that.

**Comment by podaen** [comment]
This is the popup I get when I open the terminal.


![WindowsTerminal_YJrxwiGwA9](https://github.com/kyaulabs/win11tweak/assets/63886347/2a09904d-2c79-456b-8326-6960e062ef8f)

**Comment by kyau** [comment]
The git-for-windows and msys2 environments are currently running mismatched versions of openssl (they each run a different version, they both use to be on the same version) which is causing all sorts of compatibility issues for msys2. I was able to resolve this initially during post-installation but it required a lot of personal touch, odds are this was changed again somehow which broke my hacked together method of combining them together again.

I can spin up a development copy this weekend or maybe next week some time and take a look at what is going on.

See: https://github.com/git-for-windows/git/wiki/Install-inside-MSYS2-proper

As for Agave, this is merely a monospace terminal font, the only way this does not install correctly (if I remember correctly) is if the original script is merely double-clicked on or run from Windows Terminal. As stated in the `README.md` it has to be run from a command prompt (`cmd.exe`) window.

**Comment by podaen** [comment]
Definitely git together with windows ssl! After I uninstalled GIT and used the buildin GIT in VS I saw that it was still referring to a cert bundle in to program files which I solved by using secure channel in to the VS setting.  But I still had this 'gitconfig' file in to C:\Program Files\Git\etc\

> [diff "astextplain"]
> 	textconv = astextplain
> [filter "lfs"]
> 	clean = git-lfs clean -- %f
> 	smudge = git-lfs smudge -- %f
> 	process = git-lfs filter-process
> 	required = true
> [http]
> 	sslBackend = openssl
> 	sslCAInfo = C:/Program Files/Git/mingw64/ssl/certs/ca-bundle.crt
> [core]
> 	autocrlf = true
> 	fscache = true
> 	symlinks = true
> [pull]
> 	rebase = false
> [credential]
> 	helper = manager

 VS was probably linked to this file. So I restored C:\msys64\var\lib\pacman\local\mingw-w64-x86_64-curl-winssl-8.2.1-1  and used the VS setting.


I uninstalled Agave and still had this issue, also when I install it again with chocolatery. The fond keeps been selected after uninstalling it. Not that it gave me a fatal error. but there is no fun in clicking OK every time when I open CMD or the Terminal or PowerShell. 

I found this too change the font.

https://pureinfotech.com/change-font-face-windows-terminal/

And this works for me.

**Comment by kyau** [comment]
I guess I am a bit confused as to your setup.

There is no reason to install Git like you are doing, there should not be one located in Program Files, if you used the original scripts here you would end up with an MSYS2 install with Git for Windows integrated into it and installed via pacman with no need for an external version. I then loop this very version through VS (Pro and Code).

I have some time this afternoon, I am loading up a VM to look into the conflict with MSYS2 and Git for Windows (winssl). Both of my installs (desktop and laptop) are now about a month or two old, it is possible things have changed.

**Comment by podaen** [comment]
I use VS pro. Git was already removed. Just had some residue.  I am new with pacman, but it sounds great! Want to know more about it!

What I did is install the ESP Arduino Core with GIT. Now SSL certs are loaded form this location by Windows default. I think your script is confused in what to use like VS did. 

GNU I know verry well, GCC is pretty new to me. Don't think that you need to update now for some minor changes.

**Comment by kyau** [comment]
Basically on Windows for a Linux terminal you have a few options at this point (WSL2, Cygwin, MSYS2, etc.). I just happen to go with the MSYS2 route as I feel it fits best for the scenario at hand (Windows 11, SSH, GPG, Git, Hardware Keys... all integrated).

That said, MSYS2 uses `pacman` from Arch Linux as its package manager, think of it like `winget`, `yum` or `apt-get` if you are familiar with any of these. Adding to this, the default install of Git for Windows is based on MSYS2 but does not use MSYS2 proper (ie. the version with pacman).

I followed [Install inside MSYS2 proper](https://github.com/git-for-windows/git/wiki/Install-inside-MSYS2-proper) to combine the two into MSYS2 proper, which then allows you to update Git for Windows with `pacman` instead of using the normal methods (installers, etc.).

MSYS2 in my setup by default uses `%UserProfile%` as the `/home/<user>` directory so all relevant files, `.gitconfig`, `.ssh`, etc. would all be located there.

With VS Code and Remote Explorer I set it up to use `%UserProfile%\.ssh\config-vscode`.

MSYS2 itself has ways to integrate into the IDE for use as a terminal as well: [IDEs and Text Editors](https://www.msys2.org/docs/ides-editors/)

**Comment by podaen** [comment]
Hi, took a nap. Thanks for the explanation.

> MSYS2 in my setup by default uses `%UserProfile%` as the `/home/<user>` directory so all relevant files, `.gitconfig`, `.ssh`, etc. would all be located there.

Something is different with me. I have only the config file in my .ssl document folder and Win11tweak on my desktop. 

> Directory: C:\Users\Podaen\Desktop\win11tweak
> 
> Mode                 LastWriteTime         Length Name
> ----                 -------------         ------ ----
> d----          11/08/2023    12:45                .github
> d----          11/08/2023    12:45                Modules
> d----          11/08/2023    12:45                README_media
> d----          11/08/2023    12:45                Resources
> d----          11/08/2023    12:45                Tools
> -a---          24/05/2023    13:40            124 .gitignore
> -a---          24/05/2023    13:40           5489 CODE_OF_CONDUCT.md
> -a---          24/05/2023    13:40           1693 CONTRIBUTING.md
> -a---          24/05/2023    13:40           1824 download_av.cmd
> -a---          24/05/2023    13:40          34523 LICENSE
> -a---          24/05/2023    13:40           2238 main.cmd
> -a---          24/05/2023    13:40          27513 README.md
> -a---          24/05/2023    13:40           1716 SECURITY.md
> -a---          24/05/2023    13:40           5079 user_settings.ps1ry: 

Do you make a log file from installation?

I have currently only have Linux installed. Don't know the details of it. but it contains all C and C++ versions.

![setup_pYUpnU1n36](https://github.com/kyaulabs/win11tweak/assets/63886347/fee84050-51e4-4616-93f9-14f0b4d800bb)

BTW I had the same issue with python before. Something I can't remove anymore because it is linked somewhere. The same with drivers from different versions I had to remove manually together with failing signature on OpenCL.dll, code integrity problem caused by malfunction of the LSA and a black border around the screen. I was actually planning to reinstall windows because of those problems. But I am still thinking on it, how I will do it. Don't want to come in the same situation again. Is it possible to install new software in VM with a clone of the windows installation with all the programs I have already installed at that point?

It's great that your program is checking conflicts. If there would exist something to detect conflicts in the entire Windows without installing it, that would be a even greater tool.

**Comment by kyau** [comment]
Look in just `C:\Users\Podaen` that is what I mean by `%UserProfile%` (which is just a Windows variable for the same folder). In there you should see general Linux configuration folders, these all belong to my installation of MSYS2. 

When utilizing Visual Studio the intended way with this install is to use MSYS2 as follows:

![devenv_xJwWw9gIbL](https://github.com/kyaulabs/win11tweak/assets/470450/2aae0e74-5501-43bb-803a-c422095f3c2a)

And not use command prompt at all, doing everything via Terminal in the IDE. You can even import PATHs to `fish` or `bash` or whatever shell you are using on the MSYS2 side of things. (The one above MSYS2 in the screenshot which should be there by default shows the location of the batch file with the PATHs that you will need)

Also, no sadly there is no installation log... I should put on on the docket, I do have one for my automated Linux installers.

**Comment by podaen** [comment]
Thanks to use the terminal in VS with MSYS2. It did work, but I started with your repo. how do I download some other?

I did a clean install and the [mingw-x64-x86_64-curl](https://github.com/kyaulabs/win11tweak/issues/18#top) is gone now. I will give you feedback on the installation. Impressive that you solved some issues I had before. I have some smaller remarks on how I see an installation, but that is an personal opinion. Many thanks. Dave
